tests/check_new_syntax.py

```diff
@@ -30,6 +30,28 @@ def unparse_without_tuple_parens(node: ast.AST) -> str:
     def is_dotdotdot(node: ast.AST) -> bool:
         return isinstance(node, ast.Constant) and node.s is Ellipsis
 
+    def raise_ContextManager_error(node: ast.Subscript | ast.Name, is_subscript: bool = True) -> None:
+        first_part = f"{path}:{node.lineno}: Use `contextlib.AbstractContextManager` instead of `typing.ContextManager`"
+
+        if is_subscript:
+            new_syntax = f"contextlib.AbstractContextManager[{ast.unparse(node.slice)}]"
+            error_msg = f"{first_part}, e.g. `{new_syntax}`"
+        else:
+            error_msg = first_part
+
+        errors.append(error_msg)
+
+    def raise_AsyncContextManager_error(node: ast.Subscript | ast.Name, is_subscript: bool = True) -> None:
+        first_part = f"{path}:{node.lineno}: Use `contextlib.AbstractAsyncContextManager` instead of `typing.AsyncContextManager`"
+
+        if is_subscript:
+            new_syntax = f"contextlib.AbstractAsyncContextManager[{ast.unparse(node.slice)}]"
+            error_msg = f"{first_part}, e.g. `{new_syntax}`"
+        else:
+            error_msg = first_part
```

Comment:
We can get rid of the `e.g. foo` part to simplify this a bit. The error is clear enough without it.